src/main/java/edu/harvard/iq/dataverse/api/AbstractApiBean.java

```diff
@@ -362,7 +362,16 @@ protected User findUserOrDie() throws WrappedResponse {
             return GuestUser.get();
         }
         PrivateUrlUser privateUrlUser = privateUrlSvc.getPrivateUrlUserFromToken(requestApiKey);
+        // For privateUrlUsers restricted to anonymized access, all api calls are off-limits except for those used in the UI
+        // to download the file or image thumbs
         if (privateUrlUser != null) {
+            if (privateUrlUser.hasAnonymizedAccess()) {
+                String pathInfo = httpRequest.getPathInfo();
+                if (!(pathInfo.startsWith("/access/datafile/") && !pathInfo.substring(17).contains("/"))) {
```

Comment:
I'm wary of using "17" here. will this fail if we change the access path in any way? is there another way to express this?

Reply:
"/access/datafile/".length() ?